types/atom/index.d.ts

```diff
@@ -4811,7 +4811,8 @@ export class Task {
      *  message to the child process fails.
      */
     // tslint:disable-next-line:no-any
-    send(message: any): void;
+    send(message: string | number | boolean | object | null |
+      Array<string | number | boolean | object | null>): void;
```

Comment:
Problem is, arrays of arrays are entirely reasonable, if not exactly commonplace, so this is too restrictive. And to make it "correct" we would need recursive types, which, as you mention, isn't something TypeScript can do.

Reply:
That occurred to me right after I pushed those last commits... you're right that being specific about what types of arrays to accept is problematic for the same reasons as with objects. It took me a few minutes to realize that that type union doesn't cover nested arrays, which are probably far more common than "non-contiguous" arrays with undefined values in them, anyway. I'll just go ahead and change it to `any[]`, I guess that's the best we can do.